lib/wafris/middleware.rb

```diff
@@ -24,26 +28,38 @@ def call(env)
 
       Rack::Request.ip_filter = lambda { |ip| trusted_proxies.match?(ip) }
 
-      request = Rack::Request.new(env)
+      @env = env
+      @request = Rack::Request.new(env)
+
+      return forbidden! unless allowed_request?
+
+      app.call(env)
+    end
+
+    private
+
+    attr_reader :app, :env, :request
+
+    def forbidden!
+      LogSuppressor.puts_log(
+        "[Wafris] Blocked: #{request.ip} #{request.request_method} #{request.host} #{request.url}}"
+      )
+
+      [403, {}, ['Blocked']]
+    end
 
-      if Wafris.allow_request?(request)
-        @app.call(env)
-      else
-        LogSuppressor.puts_log(
-          "[Wafris] Blocked: #{request.ip} #{request.request_method} #{request.host} #{request.url}}"
-        )
-        [403, {}, ['Blocked']]
-      end
+    def allowed_request?
+      Wafris.allow_request?(request)
     rescue Redis::TimeoutError
       LogSuppressor.puts_log(
         "[Wafris] Wafris timed out during processing. Request passed without rules check."
       )
-      @app.call(env)
+      true
```

Comment:
This is required as the intention is to continue with the request when there are exceptions.